base/applications/mspaint/mouse.cpp

```diff
@@ -340,8 +345,109 @@ struct TwoPointDrawTool : ToolBase
     }
 };
 
+typedef enum DIRECTION
+{
+    NO_DIRECTION = -1,
+    DIRECTION_HORIZONTAL,
+    DIRECTION_VERTICAL,
+    DIRECTION_DIAGONAL_RIGHT_DOWN,
+    DIRECTION_DIAGONAL_RIGHT_UP,
+} DIRECTION;
+
+#ifndef M_PI
+    #define M_PI 3.14156265
+#endif
+
+#define THRESHOULD_DEG 15
+
+static DIRECTION
+GetDirection(LONG x0, LONG y0, LONG x1, LONG y1)
+{
+    LONG dx = x1 - x0, dy = y1 - y0;
+
+    if (labs(dx) <= 8 && labs(dy) <= 8)
+        return NO_DIRECTION;
+
+    double radian = atan2((double)dy, (double)dx);
+#define DEG2RAD(degree) (((degree) * M_PI) / 180)
+#define RAD2DEG(radian) ((LONG)(((radian) * 180) / M_PI))
```

Comment:
Looks like these macros could be placed in a more "global" place e.g. where you (re)define M_PI, or in the precomp header when you include math.h :thinking:

Reply:
Moved to `"common.h"`.

Reply:
https://stackoverflow.com/questions/1727881/how-to-use-the-pi-constant-in-c

Reply:
I think there was no misunderstanding about what M_PI was all about.